datajunction-clients/python/datajunction/nodes.py

```diff
@@ -58,24 +63,61 @@ def cubes(self):
         )
 
 
-class Node(ClientEntity):  # pylint: disable=protected-access
+@dataclass
+class Node(ClientEntity):  # pylint: disable=too-many-instance-attributes
     """
     Represents a DJ node object
     """
 
     name: str
-    description: Optional[str]
     type: str
-    mode: Optional[models.NodeMode]
+    description: Optional[str] = None
+    mode: Optional[models.NodeMode] = None
     status: Optional[str] = None
-    display_name: Optional[str]
-    availability: Optional[models.AvailabilityState]
-    tags: Optional[List[TagInfo]] = None
-    primary_key: Optional[List[str]]
-    materializations: Optional[List[Dict[str, Any]]]
-    version: Optional[str]
-    deactivated_at: Optional[int]
-    current_version: Optional[str]
+    display_name: Optional[str] = None
+    availability: Optional[models.AvailabilityState] = None
+    tags: Optional[List[Tag]] = None
+    primary_key: Optional[List[str]] = None
+    materializations: Optional[List[Dict[str, Any]]] = None
+    version: Optional[str] = None
+    deactivated_at: Optional[int] = None
+    current_version: Optional[str] = None
+    columns: Optional[List[models.Column]] = None
+    query: Optional[str] = None
+
+    def to_dict(self, exclude: Optional[List[str]] = None) -> Dict[str, Any]:
+        """
+        Convert the source node to a dictionary
+        """
+        dict_ = {
+            "name": self.name,
+            "type": self.type,
+            "description": self.description,
+            "mode": self.mode,
+            "status": self.status,
+            "display_name": self.display_name,
+            "availability": self.availability,
+            "tags": [
+                tag.to_dict() if isinstance(tag, Tag) else tag for tag in self.tags
```

Comment:
This is a case of nesting where I think `dataclasses.asdict(...)` would handle on its own.

Reply:
Yes, it fails on nested objects.